projects/exui-common-lib/src/lib/components/write-address/write-address.component.spec.ts

```diff
@@ -0,0 +1,287 @@
+  const $ADDRESS_LIST = By.css('#selectAddress > .addressList');
+
+  const $MANUAL_LINK = By.css('.manual-link');
+
+  @Component({
+    selector: `write-address-field`,
+    template: `<xuilib-write-address-field [formGroup]="formGroup">
+    </xuilib-write-address-field>`
+  })
+  class WrapperComponent {
+    @ViewChild(WriteAddressFieldComponent, /* TODO: add static flag */ {})
+    public componentUnderTest: WriteAddressFieldComponent;
+    public addressField = new FormControl({});
+    public formGroup = addressFormGroup();
+  }
+
+  let addressService: AddressService;
+  let wrapperComponent: WrapperComponent;
+  let debugElement: DebugElement;
+  let fixture: ComponentFixture<WrapperComponent>;
+
+  function addressFormGroup() {
+    return new FormGroup({
+      addressLine1: new FormControl(),
+      addressLine2: new FormControl(),
+      addressLine3: new FormControl(),
+      postTown: new FormControl(),
+      county: new FormControl(),
+      postCode: new FormControl(),
+      country: new FormControl()
+    });
+  }
+
+  function buildAddress(entryNo: number): AddressModel {
+    const address = new AddressModel();
+    address.addressLine1 = `addressLine1-${entryNo}`;
+    address.addressLine2 = `addressLine2-${entryNo}`;
+    address.addressLine3 = `addressLine3-${entryNo}`;
+    address.postTown = `postTown-${entryNo}`;
+    address.county = `county-${entryNo}`;
+    address.postCode = `postCode-${entryNo}`;
+    address.country = `country-${entryNo}`;
+    return address;
+  }
+
+  function queryPostcode(postcode: string) {
+    const postcodeField = fixture.debugElement.query($POSTCODE_LOOKUP_INPUT).nativeElement;
+    postcodeField.value = postcode;
+    postcodeField.dispatchEvent(new Event('input'));
+    fixture.detectChanges();
+    debugElement.query($POSTCODE_LOOKUP_FIND).triggerEventHandler('click', null);
+    fixture.detectChanges();
+  }
+
+  beforeEach(waitForAsync(() => {
+
+    addressService = new AddressService(null);
+    TestBed
+      .configureTestingModule({
+        imports: [
+          ReactiveFormsModule,
+        ],
+        declarations: [
+          WriteAddressFieldComponent,
+          WrapperComponent,
+
+        ],
+        providers: [
+          { provide: AddressService, useValue: addressService }]
+      })
+      .compileComponents();
+
+    fixture = TestBed.createComponent(WrapperComponent);
+    wrapperComponent = fixture.componentInstance;
+    wrapperComponent.addressField.patchValue(null);
+
+    debugElement = fixture.debugElement;
+    fixture.detectChanges();
+
+  }));
+
+  it('should render only title, lookup component and manual link when address not set', () => {
+
+    expect(debugElement.query($POSTCODE_LOOKUP)).toBeTruthy();
+    expect(debugElement.query($SELECT_ADDRESS)).toBeFalsy();
+    expect(debugElement.query($MANUAL_LINK)).toBeTruthy();
+
+    /*     expect(debugElement.query($ADDRESS_COMPLEX_FIELD)).toBeTruthy();
```

Comment:
@johnbenjamin-hmcts - are all these commented-out lines still required? Should these tests be reinstated?

Reply:
@connorpgpmcelroy - could you please check and revert?

Reply:
@Daniel-Lam - I have updated the code.